kolibri/plugins/coach/assets/src/modules/classSummary/index.js

```diff
@@ -423,5 +424,8 @@ export default {
     refreshClassSummary(store) {
       return store.dispatch('loadClassSummary', store.state.id);
     },
+    refreshClassSummaryThrottled: throttle(store => {
+      return store.dispatch('refreshClassSummary');
+    }, 15000),
```

Comment:
I feel like invoking the throttle with `{leading: true, trailing: false}` produces the most easily understandable results here, but either will reduce the invocations.

Reply:
I think the default is `{leading: true, trailing: true}`, which seemed best to me because it means you'll see recent changes as soon as possible after a navigation event, but also always get the correct information eventually, after a sequence of rapid navigation.

what do you mean by "the most easily understandable results"?

Reply:
Just that the throttling behaviour you get is most easily understood (to me at least), in terms of which invocations are executed, and which skipped.

Reply:
Yeah, that's what I'd expect from `debounce` for sure. But for `throttle` I think leading and trailing is almost always more correct.

The function is called on the trailing edge _only_ if the original function was called more than once. It essentially ensures that all requests for new information are eventually accommodated, but batches groups of greater than 2.